src/components/molecules/tabs/request/response/HeadersDisplayTable.tsx

```diff
@@ -0,0 +1,59 @@
+import { AccordionGroup, Accordion, AccordionSummary, AccordionDetails, Table } from '@mui/joy';
+import { SPHeaders } from '../../../../../types/application-data/application-data';
+
+export function HeadersDisplayTable({
+	headers,
+	label,
+}: {
+	headers: Record<string, string> | SPHeaders;
+	label: 'request' | 'response';
+}) {
+	let display = <></>;
+	if (headers.__data && typeof headers.__data != 'string') {
+		if (headers.__data.length == 0) {
+			return <></>;
+		}
+		display = (
+			<>
+				{headers.__data.map(({ key, value }, index) => (
+					<tr key={index}>
+						<td>{key}</td>
+						<td>{value}</td>
+					</tr>
+				))}
+			</>
+		);
+	} else {
+		if (Object.keys(headers).length == 0) {
```

Comment:
strict equality here would be better imo

Reply:
And anywhere else == 0 is.

Reply:
length always returns a number though - it shouldn't matter right?